### PR TITLE
[REVIEW] Sparse mean

### LINKED ISSUE #2674: Implement `mean` for `cupyx.scipy.sparse.csr.csr_matrix`
This would be similar to the already existing implementation of `sum`.

This came up when trying to run Scanpy with CuPy and Dask: https://github.com/theislab/scanpy/issues/921, and @mrocklin suggested I open an issue here.

### PR DESCRIPTION
**Pull request by cjnolet**
Closes #2674

Todo list:

- [x] add tests
- [x] documenation

### CODE REVIEW

**Comment by takagi** [comment]
pfnCI, test this please.

**Comment by cjnolet** [comment]
@takagi this is ready for review whenever you have a moment.

**Comment by cjnolet** [comment]
@leofang Done. Added tests for negative axes as well.

**Comment by leofang** [comment]
? Did you push?

**Comment by cjnolet** [comment]
@leofang, Github was still asking me for my password, doh!

**Comment by leofang** [comment]
LGTM. To expedite it, let me abuse the CI 😁

Jenkins, test this please

File: cupyx/scipy/sparse/base.py [hunk 1]
```diff
@@ -330,7 +331,70 @@ def getnnz(self, axis=None):
     def maximum(self, other):
         return self.tocsr().maximum(other)
 
-    # TODO(unno): Implement mean
+    def mean(self, axis=None, dtype=None, out=None):
+        """
+        Compute the arithmetic mean along the specified axis.
```
Comment (takagi) [inline cupyx/scipy/sparse/base.py:336]:
```suggestion
        Compute the arithmetic mean along the specified axis.

```

File: cupyx/scipy/sparse/base.py [hunk 2]
```diff
@@ -330,7 +331,70 @@ def getnnz(self, axis=None):
...
+        Args:
+            axis {-2, -1, 0, 1, None}: optional
+                Axis along which the mean is computed. The default is to
+                compute the mean of all elements in the matrix
+                (i.e., `axis` = `None`).
+            dtype (dtype): optional
+                Type to use in computing the mean. For integer inputs, the
+                default is `float64`; for floating point inputs, it is the same
+                as the input dtype.
+            out (cupy.core.ndarray): optional
```
Comment (takagi) [inline cupyx/scipy/sparse/base.py:351]:
```suggestion
            out (cupy.ndarray): optional
```

File: cupyx/scipy/sparse/base.py [hunk 3]
```diff
@@ -330,7 +331,70 @@ def getnnz(self, axis=None):
...
+                Alternative output matrix in which to place the result. It must
+                have the same shape as the expected output, but the type of the
+                output values will be cast if necessary.
+
+        Returns:
+            m  (cupy.core.ndarray) : Output array of means
```
Comment (takagi) [inline cupyx/scipy/sparse/base.py:357]:
```suggestion
            m (cupy.ndarray) : Output array of means
```

**Comment by takagi** [comment]
There is a duplicated implementation of `validateaxis` in `cupyx/scipy/sparse/utils.py`. I think `sputils.validateaxis` is fine to follow SciPy and will remove `utils.validateaxis` after this PR is merged.

**Comment by takagi** [comment]
Thanks for the changes! Then, I want to ask you for removing `utils.validateaxis`. It is used only a few places.

**Comment by cjnolet** [comment]
@takagi, I've removed `util.validateaxis` and updated everything to use `sputils.validateaxis`.

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by takagi** [comment]
LGTM, thanks!

**Comment by leofang** [comment]
Thanks, @cjnolet!

**Pull request merged**